File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        std::unordered_set<int> set1(nums1.begin(), nums1.end());
        std::unordered_set<int> set2(nums2.begin(), nums2.end());

        std::vector<int> onlyInNums1;
        std::vector<int> onlyInNums2;

        for (int num : set1) {
            if (!set2.contains(num)) {
                onlyInNums1.push_back(num);
            }
        }
        for (int num : set2) {
            if (!set1.contains(num)) {
                onlyInNums2.push_back(num);
            }
        }

        return {onlyInNums1, onlyInNums2};
    }
};

File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.dart
class Solution {
  List<List<int>> findDifference(List<int> nums1, List<int> nums2) {
    final set1 = nums1.toSet();
    final set2 = nums2.toSet();

    final onlyInNums1 = set1.difference(set2).toList();
    final onlyInNums2 = set2.difference(set1).toList();

    return [onlyInNums1, onlyInNums2];
  }
}